File: src/struct_frame/boilerplate/ts/payload_types/payload_default.ts
// Payload Default (TypeScript)
// Format: [LEN] [MSG_ID] [PACKET] [CRC1] [CRC2]

import { PayloadType, PayloadConfig } from './base';

export const PAYLOAD_DEFAULT_CONFIG: PayloadConfig = {
    payloadType: PayloadType.DEFAULT,
    name: 'Default',
    hasCrc: true,
    crcBytes: 2,
    hasLength: true,
    lengthBytes: 1,
    hasSequence: false,
    hasSystemId: false,
    hasComponentId: false,
    hasPackageId: false,
    description: '[LEN] [MSG_ID] [PACKET] [CRC1] [CRC2] - Standard format with 1-byte length and CRC.'
};

File: src/struct_frame/boilerplate/py/payload_types/payload_default.py
# Payload Default
# Format: [LEN] [MSG_ID] [PACKET] [CRC1] [CRC2]
# 1-byte length, 2-byte CRC. Standard format.

from .base import PayloadType, PayloadConfig

PAYLOAD_DEFAULT_CONFIG = PayloadConfig(
    payload_type=PayloadType.DEFAULT,
    name="Default",
    has_crc=True,
    crc_bytes=2,
    has_length=True,
    length_bytes=1,
    has_sequence=False,
    has_system_id=False,
    has_component_id=False,
    has_package_id=False,
    description="[LEN] [MSG_ID] [PACKET] [CRC1] [CRC2] - Standard format with 1-byte length and CRC."
)
